airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/retriers/backoff_strategies/wait_until_time_from_header_backoff_strategy.py

```diff
@@ -0,0 +1,37 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+import time
+from typing import Optional
+
+import requests
+from airbyte_cdk.sources.declarative.requesters.retriers.backoff_strategy import BackoffStrategy
+
+
+class WaitUntilTimeFromHeaderBackoffStrategy(BackoffStrategy):
+    """
+    Extract time at which we can retry the request from response header
+    and wait for the difference between now and that time
+    """
+
+    def __init__(self, header: str, min_wait: Optional[float] = None):
+        """
+
+        :param header: header to read wait time from
+        :param min_wait: minimum time to wait for safety
+        """
+        self._header = header
+        self._min_wait = min_wait
+
+    def backoff(self, response: requests.Response, attempt_count: int) -> Optional[float]:
+        now = time.time()
+        wait_until = response.headers.get(self._header, None)
+        if wait_until is None:
```

Comment:
should this also handle the case where the header is not a valid float or number? Or is it preferable to fail? See https://github.com/airbytehq/airbyte/pull/14112

Reply:
added an optional regex we can apply on the header to extract the value https://github.com/airbytehq/airbyte/pull/14330/files#diff-5b9f4f0c653db2096b2130dac683eed96f12052732e737a65363c7156c6f53f2R12